test/hotspot/gtest/runtime/test_mutex_rank.cpp

```diff
@@ -165,4 +195,36 @@ TEST_VM_ASSERT_MSG(MutexRank, monitor_wait_rank_special,
   monitor_rank_special_minus_one->unlock();
   monitor_rank_special->unlock();
 }
+
+TEST_VM_ASSERT_MSG(MutexRank, monitor_wait_access_leaf,
+                   ".* Attempting to acquire lock monitor_rank_leaf/11 out of order with lock monitor_rank_access/1 "
+                   "-- possible deadlock") {
+  JavaThread* THREAD = JavaThread::current();
+  ThreadInVMfromNative invm(THREAD);
+
+  Monitor* monitor_rank_access = new Monitor(Mutex::access, "monitor_rank_access", false, Mutex::_safepoint_check_never);
+  Monitor* monitor_rank_leaf = new Monitor(Mutex::leaf, "monitor_rank_leaf", false, Mutex::_safepoint_check_never);
+
+  monitor_rank_access->lock_without_safepoint_check();
+  monitor_rank_leaf->lock_without_safepoint_check();
+  monitor_rank_leaf->wait_without_safepoint_check(1);
+  monitor_rank_leaf->unlock();
+  monitor_rank_access->unlock();
+}
+
+TEST_VM_ASSERT_MSG(MutexRank, monitor_wait_tty_special,
+                   ".* Attempting to acquire lock monitor_rank_special/6 out of order with lock monitor_rank_tty/3 "
+                   "-- possible deadlock") {
+  JavaThread* THREAD = JavaThread::current();
+  ThreadInVMfromNative invm(THREAD);
+
+  Monitor* monitor_rank_tty = new Monitor(Mutex::tty, "monitor_rank_tty", false, Mutex::_safepoint_check_never);
+  Monitor* monitor_rank_special = new Monitor(Mutex::special, "monitor_rank_special", false, Mutex::_safepoint_check_never);
+
+  monitor_rank_tty->lock_without_safepoint_check();
+  monitor_rank_special->lock_without_safepoint_check();
+  monitor_rank_special->wait_without_safepoint_check(1);
```

Comment:
Same as above, should lock special first and then tty one.

Reply:
Fixed.